test/test_torch.py

```diff
@@ -5960,6 +5960,52 @@ def test_tensor_grad_warnings(self):
             foo = dummy.grad
             self.assertEqual(len(w), 1)
 
+    def test_normal_shape(self):
+        for device in torch.testing.get_all_device_types():
+            tensor1 = torch.rand(1, device=device)
+            tensor4 = torch.rand(4, device=device)
+            tensor120 = torch.rand(120, device=device)
+            tensor2145 = torch.rand(2, 1, 4, 5, device=device)
+            tensor2345 = torch.rand(2, 3, 4, 5, device=device)
+            tensor2345_non_contiguous = torch.rand(2, 4, 3, 5, device=device).permute(0, 2, 1, 3)
+            tensor2345_channels_last = tensor2345.contiguous(memory_format=torch.channels_last)
+            output2345 = torch.zeros(2, 3, 4, 5, device=device)
+            output345 = torch.zeros(3, 4, 5, device=device)
+
+            # inputs have same size
+            self.assertEqual(torch.normal(tensor2345, tensor2345).size(), (2, 3, 4, 5))
+            self.assertEqual(torch.normal(tensor2345_non_contiguous, tensor2345).size(), (2, 3, 4, 5))
+            self.assertEqual(torch.normal(tensor2345, tensor2345_channels_last).size(), (2, 3, 4, 5))
+            self.assertEqual(torch.normal(tensor2345_non_contiguous, tensor2345_channels_last).size(), (2, 3, 4, 5))
+            # scalar case
+            self.assertEqual(torch.normal(tensor2345, 2).size(), (2, 3, 4, 5))
+            self.assertEqual(torch.normal(2, tensor2345).size(), (2, 3, 4, 5))
+            # inputs are expandable tensors
+            if device == 'cpu':
+                # CPU version is written in legacy code (TH), it doesn't support broadcasting
+                with self.assertRaisesRegex(RuntimeError, "inconsistent tensor"):
+                    torch.normal(tensor2345, tensor2145)
+            else:
+                self.assertEqual(torch.normal(tensor2345, tensor1).size(), (2, 3, 4, 5))
+                self.assertEqual(torch.normal(tensor2145, tensor2345).size(), (2, 3, 4, 5))
+            # inputs are non-expandable tensors, but they have same number of elements
+            self.assertEqual(torch.normal(tensor120, tensor2345).size(), (120,))
+            self.assertEqual(torch.normal(tensor2345, tensor120).size(), (2, 3, 4, 5))
+            # inputs are non-expandable tensors and they don't have same number of elements
+            with self.assertRaisesRegex(RuntimeError, "inconsistent tensor"):
+                torch.normal(tensor2345, tensor4)
+            # output and inputs are size sompatialbe
+            self.assertEqual(torch.normal(tensor2345, tensor2345, out=output2345).size(), (2, 3, 4, 5))
+            # output and inputs are not size compatible
+            # CUDA only since CPU version is written in legacy TH, it doesn't care about the size compatiable
+            if device == 'cuda':
+                with self.assertRaisesRegex(RuntimeError, "inconsistent tensor"):
```

Comment:
This checks that the following code raises a RuntimeError. It doesn't check that every line raises an error.

To do that, we'd do something like:
```
with self.assertRaisesRegex(RuntimeError, "inconsistent tensor"):
    torch.normal(tensor2345, tensor2145, out=output345)
with self.assertRaisesRegex(RuntimeError, "inconsistent tensor"):
    torch.normal(tensor2345, tensor120, out=output345)
```